eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/io/AccessionReportWriter.java

```diff
@@ -84,31 +111,65 @@ public void setAccessionPrefix(String accessionPrefix) {
+                    boolean append = true;
+                    this.contigsWriter = new BufferedWriter(new FileWriter(this.contigsOutput, append));
+
+                    // Not loading the variants might lead to duplicates. see this method's documentation
+                    this.variantsWriter = new BufferedWriter(new FileWriter(this.variantsOutput, append));
+                } else {
+                    throw new IllegalStateException(
+                            "Can not resume step safely. All temporary files from the previous execution ("
+                            + contigsOutput.getAbsolutePath() + " and " + variantsOutput.getAbsolutePath()
+                            + ") should exist to resume the step. Please delete those files and start a new job.");
+                }
+            } else {
+                // we started a new job
+                if (contigsOutput.exists() || variantsOutput.exists()) {
+                    throw new IllegalStateException(
+                            "A new job was started (did not resume a previous job) but temporary files exist ("
+                            + contigsOutput.getAbsolutePath() + " or " + variantsOutput.getAbsolutePath()
+                            + "). Please delete them and start a new job");
+                } else {
+                    boolean append = false;
+                    this.contigsWriter = new BufferedWriter(new FileWriter(this.contigsOutput, append));
+                    this.variantsWriter = new BufferedWriter(new FileWriter(this.variantsOutput, append));
+                    executionContext.put(IS_HEADER_WRITTEN_KEY, IS_HEADER_WRITTEN_VALUE);
+                }
             }
         } catch (IOException e) {
             throw new ItemStreamException(e);
         }
     }
 
-    private void writeHeader() throws IOException {
-        fileWriter.write("##fileformat=VCFv4.2");
-        fileWriter.newLine();
-        fileWriter.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO");
-        fileWriter.newLine();
+    /**
+     * Loads contig replacement mapping from a previous execution to avoid writing duplicate contig entries in the final
+     * VCF.
+     */
+    private void loadContigMappingFromTemporaryFile(File contigMappingFile) throws IOException {
+        BufferedReader contigReader = new BufferedReader(new FileReader(contigMappingFile));
+        String line;
+        while ((line = contigReader.readLine()) != null) {
+            String[] contigColumns = line.split("\t");
+            if (contigColumns.length != 2) {
+                throw new IllegalStateException("Temporary file " + contigMappingFile.getAbsolutePath()
+                                                + " doesn't have the expected format. Please delete it and "
+                                                + "start a new job.");
+            }
+            inputContigsToInsdc.put(contigColumns[0], contigColumns[1]);
```

Comment:
can you extract the `contigColumns[0]` and `contigColumns[1]` to make the code more clear?

Reply:
I did. did you do the last review on outdated code?

Reply:
No I reviewed the latest version, maybe there is some confusion because there was a similar comment before but it was in another class